src/functions/searchResults/framework/database/query-builder.ts

```diff
@@ -1,18 +1,41 @@
 import * as mysql from 'mysql2';
+import { QueryParameters } from '../../domain/query_parameters';
 
-export enum DriverDetail {
-  ApplicationReference = 'application_reference',
-  DriverNumber = 'driver_number',
-}
+export const buildDriverDetailsSearchQuery = (queryParameters : QueryParameters): string => {
+  const parameterArray = [];
+  let queries = [];
 
-export const buildDriverDetailsSearchQuery = (
-  driverDetailKey: DriverDetail,
-  driverDetailValue: string,
-): string => {
-  const template = `
-  SELECT *
-  FROM TEST_RESULT
-  WHERE ? = ?;
-  `;
-  return mysql.format(template, [driverDetailKey, driverDetailValue]);
+  if (queryParameters.startDate && queryParameters.endDate) {
+    queries.push('test_date >= ? AND test_date <= ?');
+    parameterArray.push(queryParameters.startDate);
+    parameterArray.push(queryParameters.endDate);
+  }
+  if (queryParameters.driverId) {
+    queries.push('driver_number = ?');
+    parameterArray.push(queryParameters.driverId);
+  }
+  if (queryParameters.dtcCode) {
+    queries.push('json_extract(test_result, \'$.journalData.testCentre.costCode\') = ?');
```

Comment:
Is this JSON operation really necessary? I'm assuming this is supported by our Aurora Mysql version (min MySQL 5.7.8 ?)
There should be TestCentre ID as a separate field but I'm guessing that this is not the same as costCode?!
If we still need this field to be queryable we probably need to think about extracting it to a separate field in the TEST_RESULT table (future improvement?)

Reply:
Yes to the first two questions.
I would agree with you on the last one, definitely a high priority future improvement as this feels unnecessary and wasteful.

Reply:
Yeah, it will be a full table scan because of that operation

Reply:
I'll create a sub-task to add costCode to the test result then we can refactor this when its there :)

Reply:
Yes, please please please pull all columns to be searched on out into the TEST_RESULT table and search on them!

Cost code is already in the test result - testResult.journalData.testCentre.costCode